agent/xds/version_compat.go

```diff
@@ -0,0 +1,522 @@
+package xds
```

Comment:
This is the most magical part.

Reply:
At most likely 1-2 more releases. At worst it's when our envoy support matrix looks like:
- 1.20.x (v2 deleted)
- 1.19.x (v2 deleted)
- 1.18.x (v2 deleted)
- 1.17.x (v2 opt-in)

Reply:
Maybe put that in a doc comment here for future readers.